pulp_ansible/app/tasks/collections.py

```diff
@@ -338,11 +343,22 @@ def import_collection(
     CreatedResource.objects.create(content_object=collection_version)
 
     if repository_pk:
-        repository = AnsibleRepository.objects.get(pk=repository_pk)
-        content_q = CollectionVersion.objects.filter(pk=collection_version.pk)
-        with repository.new_version() as new_version:
-            new_version.add_content(content_q)
-        CreatedResource.objects.create(content_object=repository)
+        if not lockless:
```

Comment:
You could try to see if the repository was in the locks of this task and decide that way.
WDYT?